R/plot.R

```diff
@@ -0,0 +1,157 @@
+#' Plot rolling case and death data with CFR estimates
+#'
+#' @description Plots the case and death time series used to calculate the CFR
+#' estimates in one panel and the resulting rolling naive and corrected CFR
+#' estimates in another panel.
+#'
+#' @param df_ncfr A data.frame containing the rolling naive CFR estimates, of
+#' the form returned by rolling_cfr.R
+#'
+#' @param df_ccfr A data.frame containing the rolling corrected CFR estimates,
+#' of the form returned by rolling_cfr.R
+#'
+#' @return A ggplot object of the case and death time series used to calculate
+#' the CFR estimates in one panel (top panel) and the resulting naive and
+#' corrected CFR estimates in another panel (bottom panel)
+#' @export
+#'
+#' @examples
+#' # Read in onset to death distribution for Ebola
+#' onset_to_death_ebola <- epiparameter::epidist("ebola", "onset_to_death")$pmf
+#'
+#' # Calculate rolling naive and corrected CFRs
+#' df_ncfr <- rolling_cfr(ebola1976, correct_for_delays = FALSE)
+#' df_ccfr <- rolling_cfr(ebola1976,
+#'   correct_for_delays = TRUE,
+#'   onset_to_death_ebola
+#' )
+#'
+#' # Plotting rolling estimates
+#' plot_data_and_cfr(df_ncfr, df_ccfr)
+#'
+plot_data_and_cfr <- function(df_ncfr, df_ccfr) {
+
+  # Some basic input checking
+  stopifnot(
+    "Naive rolling CFR estimates must have columns `cases` and `deaths`" =
+      (all(c("cases", "deaths", "date") %in% colnames(df_ncfr))),
+    "Corrected rolling CFR estimates must have columns `cases` and `deaths`" =
+      (all(c("cases", "deaths", "date") %in% colnames(df_ccfr)))
+  )
+
+  # set variables to NULL
+  value <- NULL
+  variable <- NULL
+  cfr_low <- NULL
+  cfr_high <- NULL
+  cfr_me <- NULL
+  type <- NULL
+
+  # putting together the data for just the first plot, to be melted
+  # into long format (for ggplot)
+  df_data_plot <- df_ncfr[, c("date", "cases", "deaths")]
+  data.table::setDT(df_data_plot)
+
+  # melting using reshape package into longer format
+  df_data_plot_long <- data.table::melt(df_data_plot, id = "date")
+
+  # plotting the top panel of the raw data
+  p_1 <- ggplot2::ggplot(
+    data = df_data_plot_long,
+    ggplot2::aes(
+      x = date, y = value,
+      colour = variable,
+      shape = variable
+    )
+  ) +
+    ggplot2::geom_line() +
+    ggplot2::geom_point() +
+    ggplot2::scale_x_date(
+      date_labels = "%b-%d\n%Y"
+    ) +
+    ggplot2::scale_shape_manual(
```

Comment:
Quite a lot of customisation here – might want to make functionality more general in future (e.g. what if want to compare 3 CFR estimation methods)?

Reply:
Agreed - will convert to issue.

Reply:
Converted to issue #16 - this could require some consideration of what the typical workflow requirements are, and thus how general this function should be.